src/theory/quantifiers/oracle_engine.h

```diff
@@ -57,13 +57,15 @@ class OracleEngine : public QuantifiersModule
   void checkOwnership(Node q) override;
   /** Identify. */
   std::string identify() const override;
-  /** Declare oracle fun */
 
-  /** Call an oracle with a set of arguments **/
-  std::string callOracle(const std::string &binary_name, 
-                         const std::vector<std::string> &argv);
+  /** get binary from oracle interface */
+  std::string getBinaryName(const Node n);
 
-  void declareOracleFun(Node f);
+  /** Call an oracle with a set of arguments **/
```

Comment:
Can we move this to a separate file?  Like `oracle_call.h / cpp` or something?

We could have a utility like OracleCaller  which caches the I/O from a single oracle.

The point here is that I'd like oracle_engine to not involve details on how the oracle is called, instead it should be just the algorithm.

Reply:
Sure. Though I imagine that, when/if we have a modularised parser that we can pull in, this utility will be a lot more concise

Reply:
moved, put it in src/util